### PR TITLE
feat: make profile a fragment

### PR DESCRIPTION
**Pull request by rjmah1**
#### Summary of changes/features added:
- made MyProfile as a fragment
- renamed initRootFrag() to getFragmentLayout()

#### Screenshots of UI changes (update with each new commit): 
- 
image

## Pull Request Checklist
- [ ] Unit test written?
- [ ] (**For later in project**) Integration tests written?
- [x] Java docs written?

### CODE REVIEW

File: client/app/src/main/java/com/example/criengine/Activities/RootActivity.java [hunk 1]
```diff
@@ -128,8 +130,17 @@ public void goToPage(int index) {
...
-     * Overrides the back button so it does not return to the previous screen.
+     * Overrides the back button so it does not return to the previous screen unless
+     * a fragment implements the IOnBackPressed interface.
      */
     @Override
-    public void onBackPressed() {}
+    public void onBackPressed() {
+        Fragment fragment = getSupportFragmentManager().findFragmentByTag("f" + viewPager.getCurrentItem());
+
+        if (!(fragment instanceof IOnBackPressed) || !((IOnBackPressed) fragment).onBackPressed()) {
```
Comment (ducklin5) [inline client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:140]: Smart

**Review by ducklin5 — APPROVED** [review]
lgtm

File: client/app/src/main/java/com/example/criengine/Activities/RootActivity.java (hunk 1 above)
Comment (rjmah1) [reply client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:140]: you like my stack overflow searches?

Comment (rjmah1) [reply client/app/src/main/java/com/example/criengine/Activities/RootActivity.java:140]: that took me a solid 40 mins to find

**Review by AdityaHarvi — APPROVED** [review]

**Pull request merged**